dnstable_convert.c

```diff
@@ -963,6 +988,13 @@ main(int argc, char **argv)
 		case 'p':
 			preserve_empty = true;
 			break;
+		case 'S':
+			store_nmsg_source = false;
+			break;
+		case 's':
+			nmsg_source_fname = strdup(optarg);
```

Comment:
Since `nmsg_source_fname` isn't modified, you should be able to save `optarg` (or `nmsg_fname` below) directly without `strdup`.